syntax = "proto2";
package vault;

message Vault {
  required Authentication authentication = 1;
  repeated Store store = 2;
}

message Store {
    required string key = 1;
    required bytes contents = 2;
}

message Authentication {
    required int32 iterations = 1;
    required bytes salt = 2;
    required bytes hmac = 3;
}

message Token {
    required Authentication authentication = 1;
    required bytes aes_key = 2;
    required bytes aes_iv = 3;
    required bytes mac_key = 4;
}
